### ISSUE TITLE
GPG Error when compiling DALI from source (using Docker builder)

### ISSUE THREAD

**Issue by mdhexplorer** [NVIDIA/DALI#2663]
Hello! I've got something wrong when I follow the step on https://docs.nvidia.com/deeplearning/dali/user-guide/docs/compilation.html#compiling-dali-from-source-using-docker-builder-recommended to build a nvdocker with DALI.

Environment:
**OS**: Red Hat Enterprise Linux Server release 7.6 (Maipo)
**docker**:
![image](https://user-images.githubusercontent.com/63660414/107198530-2ebc5100-6a30-11eb-9838-424707404778.png)
**nvdocker**:
![image](https://user-images.githubusercontent.com/63660414/107198589-3da30380-6a30-11eb-8ea1-2ac4c29b5fa0.png)
Follow the installation guide on https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/install-guide.html#docker , run docker run --rm --gpus all nvidia/cuda:11.0-base nvidia-smi , we do have the nvidia-smi console output.

**DALI**
Follow the step on https://docs.nvidia.com/deeplearning/dali/user-guide/docs/compilation.html#compiling-dali-from-source-using-docker-builder-recommended , run the build bash (actual command is " CUDA_VERSION=11.0 CREATE_WHL=YES BUILD_TF_PLUGIN=YES CREATE_RUNNER=YES PYVER=3.7  ./build.sh "), it turns out to error as below:
![image](https://user-images.githubusercontent.com/63660414/107198794-865abc80-6a30-11eb-9b90-deaa28a43490.png)

Similar error message can be found in https://github.com/NVIDIA/nvidia-docker/issues/969 .  Is that problem with the internet configuration? How to solve on RHEL7/CentOS7?

Thx!

**Comment by mdhexplorer** [comment]
More information about environment:
GPU: T4
Driver version: 450.80.02

**Comment by JanuszL** [comment]
Hi,
This problem doesn't seem to be related to DALI itself but CUDA distribution. What you can do is to play with [Dockerfile.cuda110.x86_64.deps](https://github.com/NVIDIA/DALI/blob/master/docker/Dockerfile.cuda110.x86_64.deps#L12) and just follow the suggestions in the mentioned link.

**Comment by mdhexplorer** [comment]
Hi!
Refer to your approach, I solved the initial problem. Thx!
Here's the thing. I tried to build the docker with the instruction "CUDA_VERSION=11.0 PYVER=3.7  ./build.sh" and it came up with following error:
![image](https://user-images.githubusercontent.com/63660414/107360378-cee9a700-6b10-11eb-903a-551b02e54b98.png)
DALI version is v0.28.0. I think there might be something wrong with the "docker run" command at around line 130 of docker/build.sh. How to deal with it?

**Comment by JanuszL** [comment]
By default, docker runs with the current user permissions and [mounts source code directory](https://github.com/NVIDIA/DALI/blob/master/docker/build.sh#L133) inside and creates a build dir.
Apparently, for some reason, it cannot create it. Please check the permissions and rerun the script again.

**Comment by mdhexplorer** [comment]
I changed the installation path from opt to home, and it seems that the above error is no longer reported.  A new problem appeared when entering the next step of compiling, as shown in the figure below.
![image](https://user-images.githubusercontent.com/63660414/107369246-7f10dd00-6b1c-11eb-8f9f-9db0e24b99ec.png)
![image](https://user-images.githubusercontent.com/63660414/107369288-8cc66280-6b1c-11eb-8723-ef11dd36318a.png)
According to the error message, this looks like a source code problem?

**Comment by JanuszL** [comment]
Hi,
It seems to be related to https://bugzilla.redhat.com/show_bug.cgi?id=1925204. I'm checking if I can reproduce this with the latest manylinux image.
What you can try to do in the meantime is:
```
docker pull quay.io/pypa/manylinux2014_x86_64
REBUILD_BUILDERS=YES CUDA_VERSION=11.0 PYVER=3.7 ./build.sh
```

**Comment by JanuszL** [comment]
I can confirm. The most recent manylinux doesn't work for me. It seems we need to wait a couple of days before manylinux catch with the fix. In the meantime please try:
```
docker pull quay.io/pypa/manylinux2014_x86_64:2021-01-24-91ce2b8
docker tag quay.io/pypa/manylinux2014_x86_64:2021-01-24-91ce2b8 quay.io/pypa/manylinux2014_x86_64
REBUILD_BUILDERS=YES BUILD_TF_PLUGIN=YES CREATE_RUNNER=YES PYVER=3.7 ./build.sh
```
Sorry for the troubles, but we cannot whole worlds around us.

**Comment by mdhexplorer** [comment]
Thank you very much for your previous help! My work has made significant progress!
At present, I have reached the DALI compilation stage, and the compilation work itself seems to be no problem.
I noticed that DALI was compiled using the make -j48 command.
![image](https://user-images.githubusercontent.com/63660414/107476111-30148780-6bb0-11eb-81cb-2a5d26d42721.png)
I remotely access the machine where the compilation is performed. There have been two instances of server disconnection when compiling to around 19%. Where can I change the number of compilation cores in the container?
Thanks again for your help!

**Comment by JanuszL** [comment]
Hi,
Please adjust [this line](https://github.com/NVIDIA/DALI/blob/master/docker/build_helper.sh#L95).

**Comment by mdhexplorer** [comment]
Hi! Sorry to trouble you again.
During the compilation process just now, the network error reappeared. I think this is my network problem and I will solve it myself. I observed through another command line window that the server is still continuing to execute compilation. I observe the operation through the htop command. Below is the last active process I saw.
![image](https://user-images.githubusercontent.com/63660414/107492775-c7390980-6bc7-11eb-88d0-a9c8515c65f5.png)
Is the building process over?
